generate.py

```diff
@@ -0,0 +1,108 @@
+import csv
+import datetime
+import pytz
+import math
+import numpy as np
+import timely_beliefs as tb
+from datetime import timedelta
+from timely_beliefs.beliefs.utils import load_time_series
+import pandas as pd
+from sklearn.linear_model import LinearRegression
+
+def get_observation(csv_in,current_time):
+    """
+    Returns observation at given datetime
+
+    @param csv_in : csv file containing all forecast data
+    @param current_time : datetime string to observe
+    @param model : model to use to generate new data
+    """
+    with open(csv_in) as csv_file:
```

Comment:
You already read the csv file before. This seems repetitious. Also, it's (indirectly) within a for loop.